src/Stmt.cc

```diff
@@ -1896,17 +1896,38 @@ void WhenInfo::Build(StmtPtr ws) {
     lambda = make_intrusive<LambdaExpr>(std::move(ingredients), std::move(outer_ids), "", ws);
     lambda->SetPrivateCaptures(when_new_locals);
 
+    true_const->SetLocation(cond);
+    param->SetLocation(cond);
+    one_test->SetLocation(cond);
+    two_test->SetLocation(cond);
+    empty->SetLocation(cond);
+    test_cond->SetLocation(cond);
+    do_test->SetLocation(cond);
+    else_branch->SetLocation(cond);
+    do_bodies->SetLocation(cond);
+    dummy_return->SetLocation(cond);
+    shebang->SetLocation(cond);
+    lambda->SetLocation(cond);
```

Comment:
There might be a more compact way, but this would work (and using the more verbose Obj methods rather than the specialized SetLocation helper).
```
+    for ( const auto& e : std::vector<ExprPtr>{true_const, param, one_test, two_test, lambda} )
+        e->SetLocationInfo(cond->GetLocationInfo());
+
+    for ( const auto& s :
+          std::vector<StmtPtr>{empty, test_cond, do_test, else_branch, do_bodies, dummy_return, shebang} )
+        s->SetLocationInfo(cond->GetLocationInfo());
```

Reply:
Nice! Definitely more pleasing than a whole list of calls.